### PR TITLE
Move messaging.js to NPM package

### LINKED ISSUE #9528: Create npm package for viewer messaging
to expose messaging.js

### PR DESCRIPTION
**Pull request by chenshay**
close #9528 

it is being used here: https://github.com/ampproject/amp-viewer/blob/master/src/amp-viewer-host.js

### CODE REVIEW

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import {Messaging, WindowPortEmulator, parseMessage} from './messaging';
+import {Messaging, WindowPortEmulator, parseMessage} from 'messaging';
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: Can you explain how "messaging" name here is getting resolved? Also, what will this look like when imported from the "amp-viewer" repo?

**Comment by muxin** [comment]
shouldn't we be adding milestones to issues only?

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js (hunk 1 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: it's going to be taken straight from the file itself (not from npm)

File: extensions/amp-viewer-integration/0.1/messaging/messaging.js [hunk 2]
```diff
@@ -58,9 +55,10 @@ export function parseMessage(message) {
   if (message.charAt(0) != '{') {
     return null;
   }
-  return /** @type {?Message} */ (tryParseJson(message) || null);
+  return /** @type {?Message} */ (JSON.parse(/** @type {string} */ (message)));
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/messaging/messaging.js]: This is a big change - the previous version failed silently. You should probably preserve that. Anyone can send a message and it's not guaranteed to be what we expect. So it's best to shield ourselves and overspam error logs.

File: extensions/amp-viewer-integration/0.1/test/test-amp-viewer-integration.js [hunk 3]
```diff
@@ -15,7 +15,11 @@
  */
 
 import {AmpViewerIntegration} from '../amp-viewer-integration';
-import {Messaging, WindowPortEmulator, parseMessage} from '../messaging.js';
+import {
+  Messaging,
+  WindowPortEmulator,
+  parseMessage,
+} from '../messaging/messaging.js';
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/test/test-amp-viewer-integration.js]: No ".js" in imports.

**Comment by dvoytenko** [comment]
@chenshay Could you pls also reference the PR for `amp-viewer` project in the description of this PR?

File: extensions/amp-viewer-integration/0.1/test/test-amp-viewer-integration.js (hunk 3 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/test/test-amp-viewer-integration.js]: done

File: extensions/amp-viewer-integration/0.1/messaging/messaging.js (hunk 2 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/messaging/messaging.js]: done

**Pull request merged**